openedx/core/djangoapps/models/course_details.py

```diff
@@ -271,6 +274,8 @@ def update_from_json(cls, course_key, jsondict, user):  # pylint: disable=too-ma
                 and jsondict['self_paced'] != descriptor.self_paced):
             descriptor.self_paced = jsondict['self_paced']
             dirty = True
+            # sends out the signal that course pacing has changed
+            COURSE_PACING_CHANGE.send(sender=None, course_key=course_key, course_self_paced=descriptor.self_paced)
```

Comment:
We should fire this signal when changes are actually reflected in db – i.e. after `module_store.update_item(descriptor, user.id)`